### DISCUSSION TITLE
Safari back button breaking components

### DISCUSSION THREAD

**Discussion by brandc87** [livewire/livewire#3146] [7. Bugs (archived)]
**Describe the bug**

I've noticed in Safari navigating away from the Livewire component say to another page then using the browsers back button breaks everything in a component.

It's a simple tag list but the code:

```
    public $tag = 'all';

    public function filterTag($tag)
    {
        $this->tag = $tag;

        $this->goToPage(1);
    }
```

```
wire:click="filterTag('{{ $thisTag['name'] }}')"
```

This works on the initial page load, can be tested over at [novapackages.com](novapackages.com) as the setup is similar, the tag list at the side is built with Livewire if you browse to a package detail page and use the back button it breaks the filter and search.

Working as expected in Chrome.

**To Reproduce**

1. Visit [novapackages.com](novapackages.com) using Safari
2. Click on a project to browse to the detail page
3. Use Safaris back button
4. Tag filter and search now broken

**Screenshots**

N/A

**Desktop (please complete the following information):**

Safari 13.0.5 (macOS 10.15.3)

**Comment by sitnikovalex** [comment]
can anyone help with this problem?

**Comment by calebporzio** [comment]
I just tried the steps to reproduce in Safari and everything worked great.

Re-post if the problem still persists, closing for now.

Thank you.

**Comment by sitnikovalex** [comment]
visited novapackages.com
the problem is still there, I also have it

**Comment by Plutonic3000** [comment]
Most likely, the goToPage() function uses redirects. If not using redirects, but replacing them with links in Blade templates, functionality is preserved across pages even when using the Safari back button. Otherwise, we have freezing.

**Comment by zepfietje** [comment]
I believe this is still an issue (however novapackages.com doesn't seem to show the issue anymore).

If you create a Picci at https://picci.app/create and navigate back, the creation component is completely broken.
As far as I've tested, this only occurs on Safari macOS (works correctly in Brave and Firefox).

@Plutonic3000 I'm using a redirect after creation indeed.

**Comment by zepfietje** [comment]
REOPEN

**Comment by zepfietje** [comment]
Seems related to https://github.com/livewire/livewire/issues/2603, https://github.com/livewire/livewire/issues/2114 and https://github.com/livewire/livewire/issues/2334.

https://github.com/livewire/livewire/issues/2114#issuecomment-766530813 fixes this issue for me (it's not perfect, but probably the most elegant solution).
